File: front/src/models/CommandItem.ts
import { Commande } from "./Commande";
import { Product } from "./Product";
import { ProductVars } from "./ProductVars";

export interface CommandItem {
  id: number;
  commande: Commande;
  product: Product;
  variant: ProductVars;
  quantity: number;
  unitPrice: number;
}
export interface CommandeItemDto {
  id: number;
  quantity: number;
  unitPrice: number;
  productId: number;
  productName: string;
  productRef:string;
  productImage: string | null;
  brandName: string | null;
  variantId: number;
  variantSize: string;
  variantColor: string;
}
export interface CommandeItemRequest {
  productId: number;
  variantId: number;
  quantity: number;
}

File: server/src/main/java/freelance/twin/sport/server/commande/entity/CommandItem.java
package freelance.twin.sport.server.commande.entity;

import freelance.twin.sport.server.product.entity.Product;
import freelance.twin.sport.server.product.entity.ProductVars;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "commande_items",
        indexes = {@Index(name = "idx_commande_item_commande_ref", columnList = "commande_ref")}
        )
public class CommandItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "commande_id", nullable = false)
    private Commande commande;

    @Column(name = "commande_ref")
    private String commandeRef;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @ManyToOne
    @JoinColumn(name = "variant_id", nullable = false)
    private ProductVars variant;

    private int quantity;
    private double unitPrice;
}
